### PR TITLE
CI: integration tests to mysql80

### PR DESCRIPTION
**Pull request by maxknv**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- all mysql tests migrated to mysql8.0
- new test set added test_mysql57_database_engine which is equal to test_mysql_database_engine but runs with 5.7 as a  minimal check to test compatibility with EOL mysql5.7

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit b054743d2c6b01ee8be886a53d65052154b940bc with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59939/b054743d2c6b01ee8be886a53d65052154b940bc/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59939/b054743d2c6b01ee8be886a53d65052154b940bc/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/ci/ci_config.py [hunk 1]
```diff
@@ -637,16 +637,8 @@ def validate(self) -> None:
         Labels.CI_SET_INTEGRATION: LabelConfig(
             run_jobs=[
                 JobNames.STYLE_CHECK,
-                Build.PACKAGE_ASAN,
```
Comment (rschu1ze) [inline tests/ci/ci_config.py:640]: Unrelated changes?

**Review by rschu1ze — APPROVED** [review]

File: tests/ci/ci_config.py (hunk 1 above)
Comment (maxknv) [reply tests/ci/ci_config.py:640]: just a bit related. i use #ci_set_imtegration tag to test my changes in ci. i made this tag to run only "integration tests (release)"

**Comment by maxknv** [comment]
test_backup_restore_s3/test.py::test_user_specific_auth is flaky, other tests are ok
